eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -428,6 +431,27 @@ public void testCachedBadResponseAllowsLaterFetching() {
         assertEquals("control", assignment);
     }
 
+    private void waitForPopulatedCache() {
+        long waitStart = System.currentTimeMillis();
+        long waitEnd = waitStart + 10 * 1000; // allow up to 10 seconds
+        boolean cachePopulated = false;
+        try {
+            File file = new File(ApplicationProvider.getApplicationContext().getFilesDir(), CACHE_FILE_NAME);
+            while (!cachePopulated) {
```

Comment:
Is there any possibility of this resulting in an infinite loop? If yes, maybe set a max retries?

Reply:
The next couple lines will throw an error if the 10-second timeout expires